testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -397,6 +398,111 @@ public void testDescribeTable()
         assertEquals(actualColumns, expectedColumns);
     }
 
+    @Override
+    public void testView()
+    {
+        // TODO merge AbstractTestDistributedQueries into BaseConnectorTest
+        super.testView();
+    }
+
+    @Test
+    public void testMaterializedView()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+            assertQueryFails("CREATE MATERIALIZED VIEW nation_mv AS SELECT * FROM nation", "This connector does not support creating materialized views");
+            return;
+        }
+
+        String catalogName = getSession().getCatalog().orElseThrow();
+        String schemaName = getSession().getSchema().orElseThrow();
+        String viewName = "test_materialized_view_" + randomTableSuffix();
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM nation");
+        // reading
+        assertThat(query("SELECT * FROM " + viewName))
+                .skippingTypesCheck()
+                .matches("SELECT * FROM nation");
+
+        // table listing
+        assertThat(query("SHOW TABLES"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + viewName + "'");
+        // information_schema.columns without table_name filter
+        assertThat(query("SELECT table_name FROM information_schema.tables WHERE table_schema = '" + schemaName + "'"))
+                .skippingTypesCheck()
+                .containsAll("VALUES '" + viewName + "'");
+        // information_schema.columns with table_name filter
+        checkInformationSchemaTablesForPointedQueryForMaterializedView(schemaName, viewName);
```

Comment:
also add a test which filters byt type expecting `VIEW` or `MATERIALIZED VIEW` depending on what we settle on.

Or assert on type.

Reply:
@losipiuk  i think we didn't settle on this yet, but that's a very good idea.

for now will have assertion like this:
```
.containsAll("VALUES ('" + viewName + "', 'BASE TABLE')"); // TODO table_type should probably be "* VIEW"
```